src/coreclr/pal/src/init/pal.cpp

```diff
@@ -147,6 +143,42 @@ static bool RunningNatively()
 }
 #endif // __APPLE__
 
+#if defined(__sun)
```

Comment:
Might be able to use `TARGET_SUNOS` here too.

Reply:
I think it's ok to leave it as __sun if nearby context has the similar usage.